packages/nodes-from-anchor/src/v01/AccountNode.ts

```diff
@@ -0,0 +1,31 @@
+import {
+    AccountNode,
+    accountNode,
+    bytesTypeNode,
+    camelCase,
+    fieldDiscriminatorNode,
+    structFieldTypeNode,
+    structTypeNode,
+} from '@kinobi-so/nodes';
+
+import { getAnchorDiscriminatorV01 } from './../discriminators';
+import { IdlV01Account } from './idl';
+
+export function accountNodeFromAnchorV01(idl: IdlV01Account): AccountNode {
+    const idlName = idl.name;
+    const name = camelCase(idlName);
+
+    const discriminator = structFieldTypeNode({
+        defaultValue: getAnchorDiscriminatorV01(idl.discriminator),
+        defaultValueStrategy: 'omitted',
+        name: 'discriminator',
+        type: bytesTypeNode(),
+    });
+
+    // TODO: How to set defined type link to the field definition in types list?
+    return accountNode({
+        data: structTypeNode([discriminator]),
+        discriminators: [fieldDiscriminatorNode('discriminator')],
+        name,
+    });
```

Comment:
In my understanding of defined link type that would create work for a field on the structTypeNode. 

In this case the struct type node can define the discriminator as a field but needs to concat the fields stored on the type 

```suggestion
    // TODO: How to set defined type link to the field definition in types list?
    return accountNode({
        data: structTypeNode([discriminator]),
        discriminators: [fieldDiscriminatorNode('discriminator')],
        references: defnedTypeLink(name)
        name,
    });
```

Reply:
I'll explain this in more details tomorrow morning but this is one of the only place you can't use a `DefinedTypeLinkNode` (for the account data). I may revisit this in the future but `DefinedTypeLinkNode` have the ability to link outside of the tree using the `importFrom` optional attribute. This means we could end up with an `AccountNode` with no data and explorers wouldn't know what do to with it. For now, I think it's best to copy the defined type here and ensure it is a `StructTypeNode` (throwing an error otherwise).